spec/bowling_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'bowling'
+describe Bowling do
+  it 'it responds to add_roll' do
+    expect(subject).to respond_to :add_roll
+  end
+
+  it 'add_roll adds to the total' do
+    new_game = Bowling.new
+    new_game.add_roll(6)
+    expect(new_game.total).to eq 6
+  end
+
+  it 'scoring a spare applies bonus' do
+    new_game = Bowling.new
+    new_game.add_roll(6)
+    new_game.add_roll(4)
+    expect(new_game.bonus).to eq 1
```

Comment:
Are users going to use `bonus` and `total` directly to know the overall score? Or should they be private? Depending on the answer, this is maybe testing implementation more than behaviour.

You want to first think about the interface (how the user will use the class, which methods are public), and then write you tests using on what the user could access. This is so that your tests are not locking you into a specific implementation, or specific vraiable names and type. The only thing fixed is the interface.

Reply:
I was definitely planning on using total to calculate the scores, but bonus is definitely something that they shouldn't have access to. It's odd, I've heard coaches on this course say 'dont test implementation, test the behaviour' but this is the first time that it's clicked.